common/types/offer_test.go

```diff
@@ -5,17 +5,18 @@ import (
 	"fmt"
 	"testing"
 
+	ethcommon "github.com/ethereum/go-ethereum/common"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestOffer_MarshalJSON(t *testing.T) {
-	offer := NewOffer(ProvidesXMR, 100.0, 200.0, 1.5)
+	offer := NewOffer(ProvidesXMR, 100.0, 200.0, 1.5, EthAssetETH)
 	id := offer.GetID()
 	require.False(t, id.IsZero())
 
 	expected := fmt.Sprintf(
-		`{"ID":"%s","Provides":"XMR","MinimumAmount":100,"MaximumAmount":200,"ExchangeRate":1.5}`, id)
+		`{"ID":"%s","Provides":"XMR","MinimumAmount":100,"MaximumAmount":200,"ExchangeRate":1.5,"EthAsset":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]}`, id)
```

Comment:
you can implement `MarshalJSON` on the `EthAsset` type so that it marshals to a hex string, see https://pkg.go.dev/encoding/json#example-package-CustomMarshalJSON